### PR TITLE
Restrict sms messages retry

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Restrict sms messages retry
## Impacts

The SMS signup
## NEWS file entries

New Features
++++++++++++
Allow limiting the number of times a user can request a sms message
## Delete branch after merge

NO

### CODE REVIEW

**Comment by louismunro** [comment]
Fails with this error in testing:
Jan 30 14:22:56 httpd.portal(18436) INFO: Max tries reached for requesting sms for f8:a9:d0:40:6e:0b (captiveportal::PacketFence::Controller::Signup::doSmsSelfRegistration)
Jan 30 14:22:56 httpd.portal(18436) ERROR: Deep recursion detected calling "/signup/validatePreregistration" (captiveportal::PacketFence::Controller::Root::end)
Jan 30 14:22:56 httpd.portal(18436) ERROR: Deep recursion detected calling "/signup/validateMandatoryFields" (captiveportal::PacketFence::Controller::Root::end)
Jan 30 14:22:56 httpd.portal(18436) ERROR: Deep recursion detected calling "/signup/validateByEmailSource" (captiveportal::PacketFence::Controller::Root::end)
Jan 30 14:22:56 httpd.portal(18436) ERROR: Deep recursion detected calling "/signup/validateBySponsorSource" (captiveportal::PacketFence::Controller::Root::end)
Jan 30 14:22:56 httpd.portal(18436) ERROR: Deep recursion detected calling "/signup/setupSelfRegistrationSession" (captiveportal::PacketFence::Controller::Root::end)
Jan 30 14:22:56 httpd.portal(18436) ERROR: Deep recursion detected calling "/signup/doSmsSelfRegistration" (captiveportal::PacketFence::Controller::Root::end)

**Comment by jrouzierinverse** [comment]
This has been fixed please retest

**Comment by extrafu** [comment]
Bump

**Comment by louismunro** [comment]
currently testing!

File: html/captive-portal/lib/captiveportal/PacketFence/Controller/Signup.pm [hunk 1]
```diff
@@ -386,6 +386,11 @@ sub doSmsSelfRegistration : Private {
     my $mobileprovider = $request->param("mobileprovider");
     my ($pid, $phone)  = @{$session}{qw(guest_pid phone)};
 
+    if ($self->reached_retry_limit($c, 'sms_request_retries', $portalSession->profile->{_sms_request_retries})) {
+        $logger->info("Max tries reached for requesting sms for $mac");
```
Comment (cgx) [inline html/captive-portal/lib/captiveportal/PacketFence/Controller/Signup.pm:390]: I would rephrase it like this: _Maximum number of SMS signup requests reached for $mac_

File: html/pfappserver/lib/pfappserver/Form/Config/ProfileCommon.pm [hunk 2]
```diff
@@ -44,7 +44,7 @@ The captival portal block
 
 has_block 'captive_portal' =>
   (
-    render_list => [qw(logo redirecturl always_use_redirecturl nbregpages sms_pin_retry_limit)],
+    render_list => [qw(logo redirecturl always_use_redirecturl nbregpages sms_pin_retry_limit sms_request_retries)],
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Form/Config/ProfileCommon.pm:47]: How about `sms_request_limit`? It would adopt the format of the new `sms_pin_retry_limit` parameter.

**Comment by louismunro** [comment]
Now contains less fail. 
Merging.
